File: algorithms/Factorial/Factorial.py
def factorial(n):
    fact = 1
    for i in range(1, n+1):
        fact *= i
    return fact
 
def factorial_recursion(n):
    if (n == 0):
        return 1
    else:
        return n * factorial_recursion(n-1)

number = int(input("Enter a number: "))
if num < 0:
    print("Sorry, Factorial does not exist for negative numbers")
else:
    print("Factorial using Recursive Approach: ", factorial_recursion(number))
    print("Factorial using Iterative Approach: ", factorial(number))

File: algorithms/Factorial/Factorial.swift
import Foundation

while true {
    
    print("Enter a number to factorize")
    let x = Int(readLine()!)
    let number: Int = x!
    var fact: Int = 1
    
    let n: Int = number + 1
    
    for i in 1..<n{
        
        fact = fact * i
        
    }
    
    print("Factorial of ", number ," is", fact)
    
}
